### PR TITLE
Add conversion of \r to \n in books

### PR DESCRIPTION
**Pull request by Coder12a**
Some files or editors may use \r instand of \n like notepad++. If you copy text written in notepad++ and pasted into the book. The book will only have one page.
This edit convert's \r to \n.

### CODE REVIEW

File: mods/default/craftitems.lua [hunk 1]
```diff
@@ -112,6 +112,16 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
...
+		while(true) do
+			local num_start,num_end = data.text:find("\r")
+			if num_start ~= nil and num_end ~= nil then
+				local str_start = data.text:sub(1,num_start-1)
+				local str_end = data.text:sub(num_end+1)
+				data.text = str_start .. "\n" .. str_end
+			else
+				break
+			end
+		end
```
Comment (sfan5) [inline mods/default/craftitems.lua]: you can simplify everything into `data.text = data.text:gsub("\r", "\n")`

Comment (SmallJoker) [reply mods/default/craftitems.lua]: @sfan5 that will cause an unintended additional new line for `\r\n` line endings.
`data.text = data.text:gsub("\r\n", "\n")`

**Comment by C1ffisme** [comment]
For the sake of version control, I don't think the commit should be named something as vague as "The Book.", can you rename it to something like "Replace \r with \n to support copy-pasting into books from Notepad++"?

File: mods/default/craftitems.lua (hunk 1 above)
Comment (sfan5) [reply mods/default/craftitems.lua]: I know that, what I suggested is just shorter code for what he described.

**Comment by rubenwardy** [comment]
> Add conversation of /r to /n in books

File: mods/default/craftitems.lua [hunk 2]
```diff
@@ -112,6 +112,16 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 		end
 		data.description = "\""..short_title.."\" by "..data.owner
 		data.text = fields.text:sub(1, max_text_size)
+		while(true) do
+			local num_start,num_end = data.text:find("\r")
```
Comment (DS-Minetest) [inline mods/default/craftitems.lua]: Missing space.
http://dev.minetest.net/Lua_code_style_guidelines#Spaces
(More missing spaces down there, too.)

File: mods/default/craftitems.lua [hunk 3]
```diff
@@ -112,6 +112,16 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 		end
 		data.description = "\""..short_title.."\" by "..data.owner
 		data.text = fields.text:sub(1, max_text_size)
+		while(true) do
```
Comment (DS-Minetest) [inline mods/default/craftitems.lua]: `while true do`
http://dev.minetest.net/Lua_code_style_guidelines#Misc point 4

File: mods/default/craftitems.lua [hunk 4]
```diff
@@ -112,6 +112,16 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
...
+			local num_start,num_end = data.text:find("\r")
+			if num_start ~= nil and num_end ~= nil then
```
Comment (DS-Minetest) [inline mods/default/craftitems.lua]: `if num_start and num_end then`
> Don't compare values explicitly to true, false, or nil, unless it's really needed.

http://dev.minetest.net/Lua_code_style_guidelines#Misc point 3

**Comment by C1ffisme** [comment]
Conversation means a discussion or argument, _conversion_ means to change something into another thing.

(The rest of the PR title seems fine, but it would be great if we could change the actual commit title. I'm not sure if that's possible with Github, though.)

**Comment by rubenwardy** [comment]
That was a mistake due to autocorrect on my phone :P

File: mods/default/craftitems.lua (hunk 1 above)
Comment (Coder12a) [reply mods/default/craftitems.lua]: Hmm. This works better.

**Comment by C1ffisme** [comment]
> That was a mistake due to autocorrect on my phone :P

Ah, I understand.

**Comment by paramat** [comment]
045abbd6d7bac8d4706731cb36c1c15dd98fdf13

File: mods/default/craftitems.lua [hunk 5]
```diff
@@ -112,6 +112,7 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 		end
 		data.description = "\""..short_title.."\" by "..data.owner
 		data.text = fields.text:sub(1, max_text_size)
+		data.text = data.text:gsub("\r\n", "\n"):gsub("\r", "\n")
```
Comment (srifqi) [inline mods/default/craftitems.lua]: How about compacting into one line?
```lua
data.text = fields.text:sub(1, max_text_size):gsub("\r\n", "\n"):gsub("\r", "\n")
```

**Pull request closed without merging**